### ISSUE TITLE
gqlgen only return 200 error

### ISSUE THREAD

**Issue by meghavinam** [99designs/gqlgen#2487]
I want to return Http status code errors(400,500) in gqlgen Graphql Query

I tried

    "github.com/vektah/gqlparser/v2/gqlerror"

	err := gqlerror.Error{
			Message: "SSS",
			Extensions: map[string]interface{}{
				"code": 400,
			},
		}

this method.But its return 200 APi response with this error body.

How to change this 200 error to 400 error in GQL API?

**Comment by cristiprg** [comment]
@meghavinam I agree with @FlymeDllVa, essentially the GrahpQL specification does not mention anything about the implementation, it could be REST, gRPC, your own protocols etc. Therefore, bringing REST-style specs into GraphQL isn't desirable.

You're using `Extenstions` correctly in your example, and your GraphQL client can make decisions based on the those bespoke `code`s but the GraphQL request itself is essentially successful.

That's just my opinion

**Comment by wiegell** [comment]
The framework will return 422 on non-existing queries already, so it should be possible to also implement other returncodes when throwing errors. It's confusing to not have read lines in the browser console, if you always get 200.
